lib/index.js

```diff
@@ -307,6 +312,10 @@ class ElectronDownloader {
 ElectronDownloader.tmpFileCounter = 0
 
 module.exports = function download (opts, cb) {
-  const downloader = new ElectronDownloader(opts)
-  downloader.downloadIfNotCached(cb)
+  try {
+    const downloader = new ElectronDownloader(opts)
+    downloader.downloadIfNotCached(cb)
+  } catch (err) {
+    cb(err)
+  }
```

Comment:
Oof. Would be nice if this module was ported to a Promise-based API.

Reply:
Yes, yes it would 😆 #70